src/lib.rs

```diff
@@ -299,6 +308,164 @@ impl Reflection {
+                    Self::calculate_variable_size_bytes(reflect, var_type_instruction)?;
+
+                let type_constant_count =
+                    get_operand_at!(type_instruction, Operand::LiteralInt32, 1)?;
+                Ok(type_size_bytes * type_constant_count)
+            }
+            spirv::Op::TypeArray => {
+                debug_assert!(type_instruction.operands.len() == 2);
+                let type_id = get_operand_at!(type_instruction, Operand::IdRef, 0)?;
+                let var_type_instruction =
+                    Self::find_assignment_for(&reflect.types_global_values, type_id)?;
+                let type_size_bytes =
+                    Self::calculate_variable_size_bytes(reflect, var_type_instruction)?;
+
+                let var_constant_id = get_operand_at!(type_instruction, Operand::IdRef, 1)?;
+                let constant_instruction =
+                    Self::find_assignment_for(&reflect.types_global_values, var_constant_id)?;
+                let type_constant_count =
+                    get_operand_at!(constant_instruction, Operand::LiteralInt32, 0)?;
+
+                Ok(type_size_bytes * type_constant_count)
+            }
+            spirv::Op::TypeStruct => {
+                if !type_instruction.operands.is_empty() {
+                    let byte_offset = Self::byte_offset_to_last_var(reflect, type_instruction)?;
+                    let last_var_idx = type_instruction.operands.len() - 1;
+                    let id_ref = get_operand_at!(type_instruction, Operand::IdRef, last_var_idx)?;
+                    let type_instruction =
+                        Self::find_assignment_for(&reflect.types_global_values, id_ref)?;
+                    Ok(byte_offset
+                        + Self::calculate_variable_size_bytes(reflect, type_instruction)?)
+                } else {
+                    Ok(0)
+                }
+            }
+            _ => Ok(0),
+        }
+    }
+
+    pub fn get_push_constant_range(&self) -> Result<Option<PushConstantInfo>, ReflectError> {
+        let reflect = &self.0;
+
+        let push_constants = reflect
+            .types_global_values
+            .iter()
+            .filter(|i| i.class.opcode == spirv::Op::Variable)
+            .filter_map(|i| {
+                let cls = get_operand_at!(*i, Operand::StorageClass, 0);
+                match cls {
+                    Ok(cls) if cls == spirv::StorageClass::PushConstant => Some(Ok(i)),
+                    Err(err) => Some(Err(err)),
+                    _ => None,
+                }
+            })
+            .collect::<Vec<_>>();
+
+        if push_constants.len() > 1 {
+            return Err(ReflectError::TooManyPushConstants);
+        }
+
+        let push_constant = if let Some(push_constant) = push_constants.into_iter().next() {
+            match push_constant {
+                Ok(push_constant) => push_constant,
+                Err(err) => return Err(err),
+            }
```

Comment:
```suggestion
            .collect::<Result<Vec<_>>>()?;

        if push_constants.len() > 1 {
            return Err(ReflectError::TooManyPushConstants);
        }

        let push_constant = if let Some(push_constant) = push_constants.into_iter().next() {
            push_constant
```
Otherwise multiple errors and a successful match are going to skew the results because `push_constants.len()` is not accurate anymore (includes `Some(Err(...))`).